bokehjs/test/unit/models/sources/cds_view.ts

```diff
@@ -74,12 +78,46 @@ describe("CDSView", () => {
       expect([...cds.indices]).to.be.equal([0, 1, 2])
     })
 
-    it("updates indices when filters is changed", async () => {
+    it("updates indices when filter is changed", async () => {
       const view = new CDSView({filter: filter1})
       await build(view, source)
       expect([...view.indices]).to.be.equal([0, 1, 2])
       view.filter = filter2
       expect([...view.indices]).to.be.equal([1, 2, 3])
+      view.filter = filter3
+      expect([...view.indices]).to.be.equal([0, 1, 4])
+    })
+
+    it("updates indices when filter's properties are changed", async () => {
+      const boolean_filter = new BooleanFilter({booleans: [true, true, false, false, true]})
+      const view0 = new CDSView({filter: boolean_filter})
+      await build(view0, source)
+      expect([...view0.indices]).to.be.equal([0, 1, 4])
+      boolean_filter.booleans = [false, true, true, false, true]
+      expect([...view0.indices]).to.be.equal([1, 2, 4])
+
+      const index_filter = new IndexFilter({indices: [0, 1, 2]})
+      const view1 = new CDSView({filter: index_filter})
+      await build(view1, source)
+      expect([...view1.indices]).to.be.equal([0, 1, 2])
+      index_filter.indices = [1, 2, 3]
+      expect([...view1.indices]).to.be.equal([1, 2, 3])
+
+      const union_filter = new UnionFilter({operands: [boolean_filter, index_filter]})
+      const view2 = new CDSView({filter: union_filter})
+      await build(view2, source)
+      expect([...view2.indices]).to.be.equal([1, 2, 3, 4])
+      boolean_filter.booleans = [false, false, false, false, true]
+      index_filter.indices = [0, 3]
+      expect([...view2.indices]).to.be.equal([0, 3, 4])
+
+      const inversion_filter = new InversionFilter({operand: union_filter})
+      const view3 = new CDSView({filter: inversion_filter})
+      await build(view3, source)
+      expect([...view3.indices]).to.be.equal([1, 2])
+      boolean_filter.booleans = [true, false, false, true, false]
+      index_filter.indices = [1, 2, 4]
+      expect([...view3.indices]).to.be.equal([])
```

Comment:
IMO this would be better as four separate tests

Reply:
The way this was written yes, but my intention was to interlock those views to fully test update capability. I changed this accordingly to verify that all views are updated as needed.